spec/policies/admin_only_user_profile_policy_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'rails_helper'
+
+RSpec.describe AdminOnly::UserProfilePolicy do
+
+  let(:user_1) { create(:user, email: 'user@random.com') }
+  let(:member) { create(:member_with_membership_app, email: 'member@random.com', company_number: '5562728336')}
+  let(:admin)  { create(:user, email: 'admin@shf.com', admin: true) }
+  let(:visitor) { build(:visitor) }
+
+  CRUD_ACTIONS = [:edit, :update, :become]
+
+  describe 'Admin is permitted everything' do
+
+    subject { described_class.new(admin, nil) }
+
+    CRUD_ACTIONS.each do | action |
+      it { is_expected.to permit_action action }
+    end
+  end
+
+
+  describe 'Member is forbidden everything' do
+    subject { described_class.new(member, nil) }
+
+    CRUD_ACTIONS.each do | action |
+      it { is_expected.to forbid_action action }
+    end
+  end
+
+
+  describe 'User (logged in) is forbidden everything' do
+    subject { described_class.new(user_1, nil) }
+
+    CRUD_ACTIONS.each do | action |
+      it { is_expected.to forbid_action action }
+    end
+  end
```

Comment:
Should add another test here for `:become`; it's a test _in addition to_ the CRUD actions.